src/main/java/org/kiwiproject/elk/ElkAppenderFactory.java

```diff
@@ -28,24 +29,27 @@
 
     private static final JsonHelper JSON_HELPER = JsonHelper.newDropwizardJsonHelper();
 
-    private String host;
-    private int port;
     private boolean useUdp;
     private boolean includeCallerData;
     private boolean includeContext = true;
     private boolean includeMdc = true;
     private Map<String, String> customFields = new HashMap<>();
     private Map<String, String> fieldNames = new HashMap<>();
 
+    private ElkLoggerConfigProvider elkLoggerConfigProvider = ElkLoggerConfigProvider.builder().build();
+
     @Override
     public Appender<ILoggingEvent> build(LoggerContext loggerContext,
                                          String s,
                                          LayoutFactory<ILoggingEvent> layoutFactory,
                                          LevelFilterFactory<ILoggingEvent> levelFilterFactory,
                                          AsyncAppenderFactory<ILoggingEvent> asyncAppenderFactory) {
 
-        requireNotBlank(host, "host must not be blank");
-        checkState(port > 0, "port must be greater than zero");
+        checkState(elkLoggerConfigProvider.canProvide(), "Unable to find ELK host and port from ElkLoggerConfigProvider");
```

Comment:
What exactly happens if we blow up here? What does Logback do?

Reply:
I'm not sure, it must handle it.  I attempted to trace it back but got lost.  We had exceptions being thrown before and it did NOT stop the service, it just logged it and moved on without using that appender.